pylana/shiny_dashboards.py

```diff
@@ -76,12 +75,15 @@ def upload_shiny_dashboard(self, dashboard_id: str, file: io.IOBase, **kwargs) -
 
         Args:
             dashboard_id: a string denoting the shiny dashboard
-            file: a io.Base pointing to the zipped dashboard code
+            file_path: a path pointing to the zipped dashboard file
 
         Returns:
             the response of the api call
         """
-        return self.post(f"/api/shiny-dashboards/{dashboard_id}/source", files=file, **kwargs)
+        with open(file_path, "rb") as file:
+            shiny_dashboard = {'file': open(file_path, 'rb')}
+            resp = self.post(f"/api/shiny-dashboards/{dashboard_id}/source", files=shiny_dashboard, **kwargs)
+        return resp
```

Comment:
* That's not using the `with` block correctly, the file is opened again.
* The dictionary is not exactly a shiny dashboard, the variable could be named more clearly or not set at all.